crates/red_knot_test/src/lib.rs

```diff
@@ -121,6 +140,42 @@ fn run_test(db: &mut db::Db, test: &parser::MarkdownTest) -> Result<(), Failures
         })
         .collect();
 
+    // Create a custom typeshed `VERSIONS` file if none was provided.
+    if let Some(ref typeshed_path) = custom_typeshed_path {
+        if !has_custom_versions_file {
+            let versions_file = typeshed_path.join("stdlib/VERSIONS");
+            let contents = typeshed_files
+                .iter()
+                .fold(String::new(), |mut content, path| {
+                    // This is intentionally kept simple:
+                    let module_path = path
+                        .as_str()
+                        .trim_end_matches(".pyi")
+                        .trim_end_matches("/__init__")
+                        .replace('/', ".");
+                    let _ = writeln!(content, "{module_path}: 3.8-");
```

Comment:
should this be something like this?

```suggestion
                    let default_version = PythonVersion::default();
                    let _ = writeln!(content, "{module_path}: {default_version}-");
```

though that might make upgrading our default Python version harder, I suppose? Not sure

Reply:
Also not sure. I feel like what we have is less surprising? And it follows this (now removed) comment from the "planned features" section:

> If no such file is created explicitly, one should be created implicitly including entries enabling all specified `<typeshed-root>/stdlib` files **for all supported Python versions**.

Reply:
hmm, yeah. So I guess if we wanted to make it dynamic, it should be something like `PythonVersion::minimum_supported()` rather than `PythonVersion::default()`. But we don't have that method, and I don't think it's worth adding just for this. So let's go with what you have now.